core/src/main/java/org/dozer/propertydescriptor/PropertyDescriptorFactory.java

```diff
@@ -49,8 +50,12 @@ public static DozerPropertyDescriptor getPropertyDescriptor(Class<?> clazz, Stri
         boolean isMapProperty = MappingUtils.isSupportedMap(clazz);
         if (name.equals(DozerConstants.SELF_KEYWORD) &&
             (mapSetMethod != null || mapGetMethod != null || isMapProperty)) {
```

Comment:
When I removed the `!= null` checks from the `PropertyDescriptorFactory`, it broke a few tests, for instance `MapTypeTest.testPropertyClassLevelMap()`.  I think it broke the ability for non Map backed objects to use `get` and `put` methods.